compiler/passes/convert-typed-uast.cpp

```diff
@@ -5071,6 +5090,157 @@ bool TConverter::enter(const ExternBlock* node, RV& rv) {
 
 void TConverter::exit(const ExternBlock* node, RV& rv) {}
 
+static SymExpr* makeCaseCond(TConverter& tc, TConverter::RV& rv,
+                             const uast::When* when,
+                             SymExpr* selectExpr,
+                             const uast::AstNode* cs) {
+  auto re = rv.byAst(cs);
+
+  // Grab the '==' ResolvedFunction
+  const AssociatedAction* action = nullptr;
+  for (const auto& a : re.associatedActions()) {
+    if (a.action() == AssociatedAction::COMPARE) {
+      action = &a;
+      break;
+    } else {
+      INT_ASSERT(false);
+    }
+  }
+  auto cmp = action->fn();
+
+  // TODO: create a wrapper for this kind of thing
+  const ResolvedFunction* rf = nullptr;
+  if (tc.paramElideCallOrNull(cmp, re.poiScope(), &rf)) {
+    INT_ASSERT(false && "Should not have param elided initializer call!");
+  } else if (!rf) {
+    return toSymExpr(TC_PLACEHOLDER((&tc)));
+  }
+
+  auto fn = tc.findOrConvertFunction(rf);
+  INT_ASSERT(fn);
+
+  // TODO: is there a better way to represent the return type of '=='? As-is,
+  //   it's a bit confusing for the case-expression's type in 'rv' to always be
+  //   a bool.
+  Expr* caseExpr = tc.convertExpr(cs, rv);
+  caseExpr = tc.storeInTempIfNeeded(caseExpr, {});
+
+  // TODO: handle case where passing to '==' has an associated action
+  auto call = new CallExpr(fn, selectExpr->copy(), caseExpr);
+
+  auto cond = tc.storeInTempIfNeeded(call, re.type());
+
+  return cond;
+}
+
+static SymExpr* getWhenCond(TConverter& tc, TConverter::RV& rv,
+                            const uast::When* when,
+                            SymExpr* selectExpr) {
+  if (when->numCaseExprs() == 1) {
+    auto cs = when->caseExpr(0);
+    return makeCaseCond(tc, rv, when, selectExpr, cs);
+  } else {
+    // Multiple cases should follow '||'-like short-circuiting, such that
+    // equality comparisons are not evaluated unless the preceding cases do
+    // not match.
+
+    VarSymbol* agg = new VarSymbol("_case_cond_agg", dtBool);
```

Comment:
Can we use a unnamed temp here instead? I ask because just the other day I ran into a bug where I decided to name a temp `chpl_error` and that actually caused the backend to be unable to locate the runtime function `chpl_error`. I feel like just using unnamed temps wherever possible might eliminate any chance of name conflicts.

Reply:
I think this is a case where I'd prefer to use the name to make logged AST easier to read. A quick search suggests this particular name isn't in use anywhere else.